### PR TITLE
commit exam

### PR DESCRIPTION
**Pull request by jpcosal**

### CODE REVIEW

**Comment by Almarean** [comment]
Allo

**Comment by Almarean** [comment]
@paulmichels HEY PAUL

File: Exam/src/exercice2/LogItemTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class LogItemTest {
+
+	@Test
+	void test() {
+		
+		LogItem log = new LogItem("info", "ceci est un test");
+		Date time = new Date();
+		
+		String expected = time.getYear()+"/"+time.getMonth()+"/"+time.getDay()+"-"
```
Comment (jpcosal) [inline Exam/src/exercice2/LogItemTest.java]: I would have prefered an explicit string for `expected` but ok.

File: Exam/src/exercice3/IFormeElementaire.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package exercice3;
+
+public interface IFormeElementaire {
+	public double calculAire();
+	public String toString();
```
Comment (jpcosal) [inline Exam/src/exercice3/IFormeElementaire.java]: Hmmm... What for?? Was not expected in the exam...

File: Exam/src/exercice3/CercleTest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package exercice3;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class CercleTest {
```
Comment (jpcosal) [inline Exam/src/exercice3/CercleTest.java]: Unit test for each child class! Very good!

File: Exam/src/exercice3/Jouet.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package exercice3;
+
+import java.util.ArrayList;
+
+public class Jouet {
```
Comment (jpcosal) [inline Exam/src/exercice3/Jouet.java]: I wish `Jouet` also implemented `IFormeElementaire`... But ok

File: Exam/src/exercice3/Jouet.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	
+	private IFormeElementaire[] formes;
+	private int nbFormes;
```
Comment (jpcosal) [inline Exam/src/exercice3/Jouet.java]: Is this even used somewhere??

File: Exam/src/exercice4/Article.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package exercice4;
+
+public class Article {
```
Comment (jpcosal) [inline Exam/src/exercice4/Article.java]: The important concept is `Sale`, which you partially got with your `Vente` class, not `Article`

File: Exam/src/exercice4/IBilan.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package exercice4;
+
+public interface IBilan {
```
Comment (jpcosal) [inline Exam/src/exercice4/IBilan.java]: You defined an interface here, but who uses it?

Moreover, it is not quite clear which purpose this interface serves...

**Pull request closed without merging**